bundles/org.openhab.binding.sonyprojector/src/main/java/org/openhab/binding/sonyprojector/internal/handler/SonyProjectorHandler.java

```diff
@@ -0,0 +1,692 @@
+import org.openhab.binding.sonyprojector.internal.SonyProjectorModel;
+import org.openhab.binding.sonyprojector.internal.SonyProjectorStateDescriptionOptionProvider;
+import org.openhab.binding.sonyprojector.internal.communication.SonyProjectorConnector;
+import org.openhab.binding.sonyprojector.internal.communication.SonyProjectorStatusPower;
+import org.openhab.binding.sonyprojector.internal.communication.sdcp.SonyProjectorSdcpConnector;
+import org.openhab.binding.sonyprojector.internal.communication.sdcp.SonyProjectorSdcpSimuConnector;
+import org.openhab.binding.sonyprojector.internal.communication.serial.SonyProjectorSerialConnector;
+import org.openhab.binding.sonyprojector.internal.communication.serial.SonyProjectorSerialOverIpConnector;
+import org.openhab.binding.sonyprojector.internal.communication.serial.SonyProjectorSerialSimuConnector;
+import org.openhab.binding.sonyprojector.internal.configuration.SonyProjectorEthernetConfiguration;
+import org.openhab.binding.sonyprojector.internal.configuration.SonyProjectorSerialConfiguration;
+import org.openhab.binding.sonyprojector.internal.configuration.SonyProjectorSerialOverIpConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SonyProjectorHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Wehrle - Initial contribution
+ * @author Laurent Garnier - Refactoring, poll thread for regular channels updates, new serial thing type, new channels
+ */
+@NonNullByDefault
+public class SonyProjectorHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SonyProjectorHandler.class);
+
+    private static final SonyProjectorModel DEFAULT_MODEL = SonyProjectorModel.VW520;
+    private static final long POLLING_INTERVAL = TimeUnit.SECONDS.toSeconds(15);
+
+    private @NonNullByDefault({}) ScheduledFuture<?> refreshJob;
+
+    private boolean identifyProjector;
+    private SonyProjectorModel projectorModel = DEFAULT_MODEL;
+    private SonyProjectorConnector connector = new SonyProjectorSdcpSimuConnector(DEFAULT_MODEL);
+
+    private SonyProjectorStateDescriptionOptionProvider stateDescriptionProvider;
+    private SerialPortManager serialPortManager;
+
+    private boolean simu;
+
+    private final Object commandLock = new Object();
+
+    public SonyProjectorHandler(Thing thing, SonyProjectorStateDescriptionOptionProvider stateDescriptionProvider,
+            SerialPortManager serialPortManager) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channel = channelUID.getId();
+        if (command instanceof RefreshType) {
+            logger.debug("REFRESH command from channel {} is ignored; will be handled by next projector state poll",
```

Comment:
How about using the `ExpiringCache` instead so that you have a chance to immediately provide a state update?

Reply:
When polling refreshed data, there is one command sent for each channel. I only run the command for linked channels. No need to send commands for information the user don't care. So I don't want to send many commands every 15 seconds just to put the value in cache in case a channel is linked during the next coming 15 seconds.
In case the user is linking a new channel, he will just wait between 0 and 15 seconds to have a data. That looks very acceptable and avoid a big traffic for nothing.

Reply:
Not sure I fully understand your answer. In general, the REFRESH command can be sent at any time, e.g. through rules, it is not just about linking new channels. The expectation is that the binding provides a state update asap - 15 seconds is maybe acceptable for exceptional situations, but definitely not ideal.

I also do not see how a cache would increase any load - the expectation would be that the cache only holds the data of what has been requested before, i.e. only for linked channels. If you receive a REFRESH for a channel that wasn't linked before (and hence you do not have a value for it in the cache), it is fine if you provide the data after executing the next polling.

Reply:
For my information, what are the actions that are triggering the REFRESH command ? What in a rule is doing that ?
I thought it was only the channel linking.

You're right, I could cache only the values I am polling.
Is there any reference binding using ExpiringCache for that kind of thing that I could look at ?

Reply:
> what are the actions that are triggering the REFRESH command ? 

REFRESH is just a command line ON and OFF. What I meant is that users could have ruled that send a REFRESH command, because they want to receive a state update for the item.

I don't have any reference binding at hand (and as my IDE is too slow, I don't have all bindings in the workspace so that I cannot search atm).

I'd be fine to merge it without such a cache, but it would be nice if you'd have a look at whether it can be improved by a follow-up PR.

Reply:
Ok thank you, for sure I will study such an enhancement in a next PR.